internal/serverinstall/nomad.go

```diff
@@ -813,29 +853,61 @@ func waypointNomadJob(c nomadConfig, rawRunFlags []string) *api.Job {
 	grpcPort, _ := strconv.Atoi(defaultGrpcPort)
 	httpPort, _ := strconv.Atoi(defaultHttpPort)
 
-	// Include services to be registered in Consul. Currently configured to happen by default
+	// Include services to be registered. Currently configured to happen with Consul by default
 	// One service added for Waypoint UI, and one for Waypoint backend port
-	if c.consulService {
+	var services []*api.Service
+	if (c.serviceProvider == "consul" && (c.consulService || upgrade)) || (c.consulService && (c.serviceProvider == "consul" || upgrade)) {
```

Comment:
This line is confusing for me so I tried to simplify it. I haven't been successful yet but I did find an edge case I wanted to ask about; could `serviceProvider` be `"nomad"`, `consulService` be `true`, while also `upgrade` is `true`? If so, then this top block will execute and not the `if c.serviceProvider == "nomad"` one.

See https://go.dev/play/p/3lLyP9eqHa1

Reply:
@catsby yes, the top block would indeed execute. However since `consulService` is false by default _for upgrades only_, a user would have to explicitly set `consulService` to true, & set `serviceProvider` to `nomad`. In light of @evanphx's earlier feedback for supporting the old configuration options, I felt that it was "okay" to favor the old flag being set over the new one.

Maybe it would make sense to print out some kind of warning or prompt for input if the input options are colliding...
```go
if c.consulService && c.serviceProvider == "nomad" {
  // warn the user this combo isn't valid
}
```

Reply:
I don't have operational experience here so I simply posed the question of whether it's valid or not after I noticed the possibility. I'll defer to you for the decision on if it's correct behavior or not 👍